### PR TITLE
Bot 1.1.0 

### LINKED ISSUE #23: Add option to remove dungeon override
Add an option that removes all overridden settings for a base dungeon.

### LINKED ISSUE #27: Add ability to remove punishments from a person's records.
Because some people enjoy warning others thousands of times for no reason

### LINKED ISSUE #28: If the bot gets kicked from a guild while there are people (timed) suspended or muted, the bot may crash
This is because the (cached) guild no longer exists to the bot

### LINKED ISSUE #31: Current staff role changes are not reflected
So, custom staff roles don't get or remove the team role when said custom staff roles are added/removed from a user

### LINKED ISSUE #47: Update whenever the bot says he or she
Update it to "they" instead. I might submit a pr tomorrow changing it myself depending if I have time

### LINKED ISSUE #77: Yoink command: only accept voice channel as arg

### LINKED ISSUE #80: Quota summary file cleanup
Inconsistent formatting:
```
	Roles: [Security, Verified Raider, Team, Administrator, Bad Dev, Suspended, Test Role, Illiterate, Developer, @everyone]
	Breakdown:
		- ModmailRespond: 6 PTS (2)			[1/11/2022, 9:52:49 AM] Logged 1 QTY.
			[1/13/2022, 2:21:28 AM] Logged 1 QTY.
		- ManualVerify: 6 PTS (2)			[1/11/2022, 9:56:20 AM] Logged 1 QTY.
			[1/11/2022, 9:56:21 AM] Logged 1 QTY.
		- NameAdjustment: 14 PTS (7)			[1/11/2022, 10:03:58 AM] Logged 1 QTY.
			[1/11/2022, 10:04:24 AM] Logged 1 QTY.
```

Also, make it less cramped so it's easier to read? And sort by completion?

### LINKED ISSUE #104: Bugged embed for /configchannel
Bugged embed on edit lgging channels, doesnt update when putting in channels. 
Also Sectionsuspend is listed twice.

### LINKED ISSUE #106: Allow to start a new afk check in the same vc as alternative to end raid
Allow chaining afks in the same vc and have the option next to end raid.
This would allow the vc to be reused for faster resets. 
Logging window should pop up anyway like with end raid.

### LINKED ISSUE #107: Leader feedback channel should be open for 10 minutes instead of 1
Leader feedback channels should stay open for 10 minutes instead of one after the raid has ended.
This is to give more time for raiders to find and write smthg meaningful.
Also there should be an option for the leader to close it with a button or command, if they feel all has been said or a troll is trying to use it.

### LINKED ISSUE #117: Location button for deaf or no vc role people during the afk phase (not pre-afk)
This is a button to get the location, only accessible for certain roles which are configured namely:
- deaf
-  no-vc role (dungeoneer role)
This button would only be added to the afk phase, when everybody is getting dragged in. (Not pre-afk since it isnt meant to be early location.)

### LINKED ISSUE #119: Start AFK checks/headcounts in pre-existing VC
> It would be vastly preferable if there was a way to initiate another AFK in a channel that already existed, rather than having to clean a channel and then remake it every time we do another run. Fungal runs are significantly faster than O3s and even Voids in dungeoneer, and we often only do 1 key then reset into more keys. 

- dooule (Fungal)

### LINKED ISSUE #123: Find command further enhancements
By default, finduser should give a very simple summary of the user in question. Only when it is requested should it give more detailed info.

### LINKED ISSUE #125: IGuildInfo (possibly) causing numerous issues with TypeScript's type checking system
Getting errors like
image

and
image

along with *very slow* type-checking, suggestions, and linting (TS Server very slow essentially). Note that as soon as `IGuildInfo` is removed, the TS Server becomes very fast. 

Major issue, needs to be fixed asap.

### LINKED ISSUE #130: Modmail blacklist, unblacklist

### PR DESCRIPTION
**Pull request by ewang2002**
NOTE: In this PR, a lot of files were changed. For a majority of these files, only minor things were changed (styling of code) and are not important. That being said, feel free to see if this causes an issue.

This PR adds the following:
- SetStatus command for developers (ability to change the bot's presence and activity)
- Ability to start a new AFK check in the same VC as an active AFK check (essentially just restarts the raid). Closes #106
- Ability to select a pre-existing VC for a raid. Closes #119 
- Gives mods the ability to allow/disallow vc selection on a per-section basis
- Add a simplified find summary, which is shown by default when the finduser command is performed. Closes #123
- Adds the ability to remove (inactive) punishments from moderation history. Closes #27
- Ability to reset overrides for a dungeon. Closes #23 
- Adds the ability to blacklist or unblacklist someone from modmail (closes #130)
- Adds early location role customization (closes #117)
- Adds universal early location reactions (i.e. early location reactions that appear on *every* dungeon's AFK checks, like Nitro).

This PR updates the following:
- Some of the core packages have been updated to the latest version.
- The find command no longer needs two separate arguments for Discord ID or IGN; it will now take in one argument that is either the Discord ID or IGN (like the punishment commands). 
- The /who raw parse results (i.e. what the bot was able to parse from the screenshot) are displayed on every parse summary embed.
- The feedback channel will now stay for 10 minutes instead of 1 minute (closes #107). 
- CleanVC command now only allows a VC as an argument (enforced by Discord's slash command feature)
- Purge command now only allows integers between 0 and 1000 (enforced by Discord's slash command feature)
- BaseCommand now allows you to add restrictions to some argument types
- Yoink command only allows VC as slash cmd argument. Closes #77. 

This PR fixes the following:
- Fixed a bug where using the find command would sometimes cause the bot to think for a long time. 
- Possibly fixed a bug where the bot will still try to send messages to the raid log channel 
- Fixes #104 
- Fixes #125
- Fixes #80
- Fixes #31
- Fixes #47 
- Fixes #28. 
- Fixed a bug where editing the early location amount for some reaction from >0 early location amount to another >0 early location amount would result in the bot thinking that you were adding another early location reaction.

### CODE REVIEW

File: src/commands/owner/SetStatus.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import {ArgumentType, BaseCommand, ICommandContext, ICommandInfo} from "../BaseCommand";
```
Comment (dmansdman) [inline src/commands/owner/SetStatus.ts:1]: I didn't have perms on testing server, could not test.

File: src/instances/HeadcountInstance.ts [hunk 2]
```diff
@@ -82,58 +90,46 @@ export class HeadcountInstance {
     private readonly _controlPanelChannel: TextChannel;
     // The section.
     private readonly _raidSection: ISectionInfo;
+    // All essential options (options that give early location). Equivalent to _afkCheckButtons but as raw data
```
Comment (dmansdman) [inline src/instances/HeadcountInstance.ts:93]: weird comment order?

Comment (dmansdman) [reply src/instances/HeadcountInstance.ts:93]: image

File: src/instances/RaidInstance.ts [hunk 3]
```diff
@@ -170,44 +174,31 @@ export class RaidInstance {
...
-
+    // A collection that contains the IAfkCheckReaction.mapKey as the key and the members with the corresponding
     // Buttons to display on the AFK check. These should only contain essential buttons.
     private readonly _afkCheckButtons: MessageButton[];
-    // All essential options (options that give early location). Equivalent to _afkCheckButtons but as raw data
+    // A collection that deals with *general* (Nitro, Patreon, etc.) early location. The key is the mapKey and the
     // instead of buttons. The key is the mapping key.
     private readonly _allEssentialOptions: Collection<string, ReactionInfoMore>;
-    // A collection that contains the IAfkCheckReaction.mapKey as the key and the members with the corresponding
     // item as the value.
```
Comment (dmansdman) [inline src/instances/RaidInstance.ts:186]: Same issue with comment order.  More stacked single-line comments that dont appear to match what they describe.  If these are intended to be multi-line comments, would make them so. and double check the wording.

Same for HeadcountInstance

File: src/instances/HeadcountInstance.ts (hunk 2 above)
Comment (ewang2002) [reply src/instances/HeadcountInstance.ts:93]: Oh wtf

**Review by dmansdman — CHANGES_REQUESTED** [review]
See comments.  In short:

- comment changes for Raid/Headcount Instance are not ideal
- I can't test setStatus

File: src/instances/RaidInstance.ts (hunk 3 above)
Comment (ewang2002) [reply src/instances/RaidInstance.ts]: should be acknowledged

File: src/commands/raid-leaders/common/RaidLeaderCommon.ts [hunk 4]
```diff
@@ -1,27 +1,26 @@
 import {
```
Comment (dmansdman) [inline src/commands/raid-leaders/common/RaidLeaderCommon.ts:1]: For the new VC system, it NEEDS to be more clear to raiders which channel is being used, namely from the AFK embed.  I know it says it in the AFK stage, I would include it in the Pre-afk stage as well.

I'm not sure how I feel about it as a whole, but maybe some people will find it useful.  For Dungeoneer, please don't add more than 2 raiding channels.

File: src/commands/staff/RemovePunishmentHistory.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import {ArgumentType, BaseCommand, ICommandContext, ICommandInfo} from "../BaseCommand";
```
Comment (dmansdman) [inline src/commands/staff/RemovePunishmentHistory.ts:1]: I'm not sure what the deal is with this command.  Either it is misnamed, or misimplemented.

removepunishment should take a moderation id and remove a single punishment
removepunishmenthist should take a NAME and remove ALL their punishments.  That's what a history is.

Also typo under cmdCode

**Comment by dmansdman** [comment]
Overall:

1) I would split the removePunishmentHist command into two parts, as mentioned in a comment above.  Also the typo while you're at it
2) Minor adjustment to afk check announcements to accommodate changes to VC. Essentially make the VC public-facing in pre-afk stage and potentially post-afk stage.
3) I feel like there are a LOT of changes in this patch that are about to be user-tested.  We will have to be receptive to potential bugs that can come up, user feedback on the changes, etc.  I recommend an announcement to the servers describing the change at a high level, what to test and provide feedback for, etc.

(1) is the only thing stopping me from approving.  the other two are nice-to-haves.

File: src/commands/raid-leaders/common/RaidLeaderCommon.ts (hunk 4 above)
Comment (ewang2002) [reply src/commands/raid-leaders/common/RaidLeaderCommon.ts:1]: Resolved

File: src/commands/staff/RemovePunishmentHistory.ts (hunk 5 above)
Comment (ewang2002) [reply src/commands/staff/RemovePunishmentHistory.ts]: It should be the former. We're just removing one entry from the punishment history, not every entry.

**Comment by ewang2002** [comment]
For (1): This was meant to be one command.

For (2): Done.

For (3): Yep.

**Review by dmansdman — APPROVED** [review]
Looks good

**Pull request merged**